src/lib/fields_derivers_zkapps/fields_derivers_zkapps.ml

```diff
@@ -355,10 +355,15 @@ module Make (Schema : Graphql_intf.Schema) = struct
         ~of_string:sign_of_string
     in
     let ( !. ) = ( !. ) ~t_fields_annots:Currency.Signed_poly.t_fields_annots in
-    Currency.Signed_poly.Fields.make_creator obj ~magnitude:!.amount
-      ~sgn:!.sign_deriver
-    |> finish "BalanceChange"
-         ~t_toplevel_annots:Currency.Signed_poly.t_toplevel_annots
+    let balance_change obj =
```

Comment:
```suggestion
    let balance_change obj' =
```

NIT: let's avoid variable name clash
If there's a better var name then `obj'`, that would be ideal